Doc/library/types.rst

```diff
@@ -409,7 +409,7 @@ Additional Utility Classes and Functions
                return "{}({})".format(type(self).__name__, ", ".join(items))
 
            def __eq__(self, other):
-               return self.__dict__ == other.__dict__
+               return self.__dict__ == other.__dict__ and type(self) == type(other)
```

Comment:
Two things:
- The actual test in the C code is `isinstance(self, SimpleNamespace) and isinstance(other, SimpleNamespace)`.  Since this would make a difference with subclasses, we should have the same test here.
- I know this is a nit, but I suggest putting the `isinstance` checks first. They should be cheap, and that's usually how code like this orders the tests.

For reference, the C version is at https://github.com/python/cpython/blob/f97406be4c0a02c1501c7ab8bc8ef3850eddb962/Objects/namespaceobject.c#L163.

Reply:
Actually, the C code is:
```
if isinstance(self, SimpleNamespace) and isinstance(other, SimpleNamespace):
   return self.__dict__ == other.__dict__ 
return NotImplemented
```
I don't know if we want to be that exact, though.

Reply:
Thank you very much for the feedback!

I had a look at the C implementation, but I still did not know whether `PyObject_TypeCheck` refers to `type` or `isinstance` - I just should have dug a bit deeper.

And thanks for pointing out to put the `isinstance` checks first. This makes a lot of sense. I learned something new!

Reply:
I just saw your followup after I pushed the suggested changes.

It is your call which solution you prefer.

Reply:
As @asvetlov also likes to see `NotImplemented`, I just updated the code.